packages/protocol/engine/tangle/blockdag/blockdag.go

```diff
@@ -131,6 +131,10 @@ func (b *BlockDAG) markSolid(block *Block) (err error) {
 		return err
 	}
 
+	if err := b.checkCommitmentMonotonicity(block); err != nil {
```

Comment:
The code is cleaner by doing this checks one after the other, but you are re-iterating over the parents and loading them in both checks. It would be more efficient to do a `checkParents` function and do the steps inside the common loop one after the other.